### PR TITLE
Skip negative or unconfirmed cases for Mexico

### PR DESCRIPTION
**Pull request by abhidg**
Mexico now reports negative cases, as well as status of RT-PCR and antigen tests. These are combined into a categorical field CLASIFICACION_FINAL. This commit uses this categorical field to skip negative or unconfirmed cases.
    
Also adds a boolean field INDIGENA which indicates whether the person self-identifies as indigenous. This is added to the notes field.
    
Currently, these new fields are ignored: RESULTADO_ANTIGENO, TOMA_MUESTRA_ANTIGENO, RESULTADO_LAB and TOMA_MUESTRA_LAB. These replace the earlier RESULTADO field.
    
Fixes: #1938

### CODE REVIEW

**Review by jim-sheldon — APPROVED** [review]

File: ingestion/functions/parsing/mexico/mexico.py [hunk 1]
```diff
@@ -160,6 +162,16 @@ def parse_cases(raw_data_file: str, source_id: str, source_url: str):
...
         for row in reader:
+            classification = parsing_lib.safe_int(row.get('CLASIFICACION_FINAL', None))
+            # 1	CASO DE COVID-19 CONFIRMADO POR ASOCIACIÓN CLÍNICA EPIDEMIOLÓGICA
+            # 2	CASO DE COVID-19 CONFIRMADO POR COMITÉ DE  DICTAMINACIÓN
+            # 3	CASO DE SARS-COV-2  CONFIRMADO
+            # 4	INVÁLIDO POR LABORATORIO
+            # 5	NO REALIZADO POR LABORATORIO
+            # 6	CASO SOSPECHOSO
+            # 7	NEGATIVO A SARS-COV-2
+            if classification is None or classification not in [1, 2, 3]:
```
Comment (iamleeg) [inline ingestion/functions/parsing/mexico/mexico.py:173]: `classification not in [1, 2, 3]` includes `classification is None`, so the first test is redundant.

File: ingestion/functions/common/parsing_lib.py [hunk 2]
```diff
@@ -41,6 +41,19 @@
...
+    if isinstance(x, float):
+        return int(x)
+    elif isinstance(x, int):
+        return x
+    elif isinstance(x, str):
+        if x.isdecimal():  # isdigit() also matches superscripts and symbols with a digit
+            return int(x)
+        else:
+            return None
+
```
Comment (iamleeg) [inline ingestion/functions/common/parsing_lib.py:56]: could be simplified to:

```
def safe_int(x):
    try:
        return int(x)
    except:
        return None
```

**Review by iamleeg — APPROVED** [review]
Looks good, just a couple of suggestions to make it simpler

**Review by fojackson8 — COMMENTED** [review]
Looks good to me, thanks!

**Pull request merged**